app/models/entity_category.rb

```diff
@@ -0,0 +1,5 @@
+class EntityCategory < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :category_items, dependent: :destroy
+  has_many :group_categories, through: :category_items
```

Comment:
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=You%20should%20validate%20all%20user%20input%20to%20make%20sure%20that%20anyone%20with%20bad%20intentions%20cannot%20compromise%20your%20app.), Please add validations to validate all user input (`:name`, `:amount`, and `:categories`) to make sure that anyone with bad intentions cannot compromise your app.